File: sort/merge_sort/java/MergeSortTest.java
import org.junit.Test;

import java.util.Arrays;

public class MergeSortTest {

	@Test
	public void test() {
        int[] list = {5, 7, 231, 606, 1, 1, 1, 6, 2, 1, 7, 3};
        int[] expectedList = {1, 1, 1, 1, 2, 3, 5, 6, 7, 7, 231, 606};
		assert(Arrays.equals(MergeSort.sort(list), expectedList));
	}

	@Test
	public void testNull() {
		int[] list = {};
		int[] expectedList = {};
		assert(Arrays.equals(MergeSort.sort(list), expectedList));
	}
}

File: sort/merge_sort/haskell/MergeSortTest.hs
module MergeSortTest.Main (main) where
import Data.List (sort)

import MergeSort (mergeSort)

main :: IO ()
main = do
    if test
    then putStrLn "List sorted correctly"
    else putStrLn "Error: List did not sort correctly"

test :: Bool
test = sort list == mergeSort list
  where
    list = [12,3,55,1,2,7,4]
